src/adera/Machines/Rocket.h

```diff
@@ -129,20 +137,28 @@ class MachineRocket : public osp::active::Machine
     osp::active::WireInput m_wiGimbal   { this, "Gimbal"   };
     osp::active::WireInput m_wiIgnition { this, "Ignition" };
     osp::active::WireInput m_wiThrottle { this, "Throttle" };
+    std::vector<ResourceInput> m_resourceLines;
 
     osp::active::ActiveEnt m_rigidBody  { entt::null };
-    fuel_list_t m_resourceLines;
     Parameters m_params;
 
     // Rocket power output for the current frame
     float m_powerOutput{0.0f};
 }; // MachineRocket
 
-inline MachineRocket::MachineRocket(Parameters params, fuel_list_t& resources)
+inline MachineRocket::MachineRocket(Parameters params, std::vector<input_t> resources)
     : Machine(true)
     , m_params(params)
-    , m_resourceLines(std::move(resources))
-{ }
+{
+    for (input_t& input : resources)
```

Comment:
I recommend signifying that you're planning on moving out of the items in the vector by using std::move() in the for loop like this:

for(input_t && input : std::move(resources))

or if that doesn't compile, then either


for(auto && input : std::move(resources))

or



for(input_t & input : std::move(resources))